frontend/javascripts/oxalis/view/right-menu/meshes_view.js

```diff
@@ -243,49 +209,146 @@ class MeshesView extends React.Component<
+    };
+
+    const loadMeshFromFile = async fileName => {
+      if (fileName === "") {
+        Toast.error("Please select a mesh file.");
+        return;
+      }
+      const pos = getPosition(this.props.flycam);
+      const id = getIdForPos(pos);
+      const layerName =
+        this.props.segmentationLayer.fallbackLayer || this.props.segmentationLayer.name;
+      const availableChunks = await getMeshfileChunksForSegment(
+        this.props.dataset.dataStore.url,
+        this.props.dataset,
+        layerName,
+        fileName,
+        id,
+      );
+      for (const chunkPos of availableChunks) {
+        // eslint-disable-next-line no-await-in-loop
+        const stlData = await getMeshfileChunkData(
+          this.props.dataset.dataStore.url,
+          this.props.dataset,
+          layerName,
+          fileName,
+          id,
+          chunkPos,
+        );
+        const geometry = parseStlBuffer(stlData);
+        getSceneController().addIsosurfaceFromGeometry(geometry, id);
+      }
+      this.props.addPrecomputedMesh(id, pos, fileName);
+    };
+
+    const handleMeshFileSelected = async mesh => {
+      loadMeshFromFile(mesh.key);
+      this.setState({ currentMeshFile: mesh.key });
+    };
+
+    const getMeshFilesDropdown = () => (
+      <Menu onClick={handleMeshFileSelected}>
+        {this.state.meshFiles.map(meshFile => (
+          <Menu.Item
+            key={meshFile}
+            value={meshFile}
+            disabled={this.state.currentMeshFile === meshFile}
+          >
+            {meshFile}
+          </Menu.Item>
+        ))}
+      </Menu>
     );
+
+    const getLoadPrecomputedMeshButton = () => (
+      <DropdownButton
+        size="small"
+        onVisibleChange={updateMeshFileList}
+        onClick={() => loadMeshFromFile(this.state.currentMeshFile)}
+        overlay={getMeshFilesDropdown()}
+        icon={<DownOutlined />}
+        buttonsRender={([leftButton, rightButton]) => [
+          <Tooltip
+            key="tooltip"
+            title={
+              this.state.currentMeshFile
+                ? `Load mesh for centered cell from file ${this.state.currentMeshFile}`
+                : "Please select a mesh file first."
+            }
+          >
+            {React.cloneElement(leftButton, {
+              style: { borderRightStyle: "dashed" },
+            })}
+          </Tooltip>,
+          React.cloneElement(rightButton, {
+            style: { borderLeftStyle: "dashed" },
+          }),
+        ]}
+      >
+        Load from Meshfile
+      </DropdownButton>
+    );
+
     const getMeshesHeader = () => (
-      <div style={{ marginTop: 10 }}>
+      <React.Fragment>
         Meshes{" "}
-        <Tooltip title="Meshes are rendered alongside the actual data in the 3D viewport. They are imported from STL files.">
+        <Tooltip title="Meshes are rendered alongside the actual data in the 3D viewport. They are imported from STL or hdf5 files. They can also be computed.">
```

Comment:
```suggestion
        <Tooltip title="Meshes are rendered alongside the actual data in the 3D viewport. They can be computed ad-hoc or pre-computed.">
```

since we don't support dropping stls anymore, the file formats seem like an unnecessary detail.